core/src/main/java/nl/nn/adapterframework/doc/model/FrankDocModel.java

```diff
@@ -0,0 +1,209 @@
+package nl.nn.adapterframework.doc.model;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.Logger;
+import org.springframework.core.annotation.AnnotationUtils;
+
+import lombok.Getter;
+import lombok.Setter;
+import nl.nn.adapterframework.doc.IbisDoc;
+import nl.nn.adapterframework.doc.IbisDocRef;
+import nl.nn.adapterframework.util.LogUtil;
+
+public class FrankDocModel {
+	private static final String JAVA_STRING = "java.lang.String";
+
+	private static Logger log = LogUtil.getLogger(FrankDocModel.class);
+
+	private @Getter List<FrankDocGroup> groups;
+	private @Getter Map<String, FrankElement> allElements = new HashMap<>();
+
+	public FrankDocModel() {
+		groups = new ArrayList<>();
+	}
+
+	public FrankElement findOrCreateFrankElement(Class<?> clazz) {
+		if(allElements.containsKey(clazz.getName())) {
+			return allElements.get(clazz.getName());
+		}
+		Class<?> superClass = clazz.getSuperclass();
+		FrankElement parent = superClass == null ? null : findOrCreateFrankElement(superClass);
+		FrankElement current = new FrankElement(clazz, parent);
+		current.setAttributes(createAttributes(clazz.getDeclaredMethods(), current));
+		allElements.put(current.getFullName(), current);
+		return current;
+	}
+
+	List<FrankAttribute> createAttributes(Method[] methods, FrankElement attributeOwner) {
```

Comment:
De code ziet er vrij complex uit, is het niet een stuk handig als je [Spring BeanUtils#getPropertyDescriptors()](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/BeanUtils.html#getPropertyDescriptors-java.lang.Class-) gebuikt om een lijst met alle `PropertyDescriptors` terug te krijgen waar je over kan itereren?

Reply:
Op een gegeven moment willen we deze code compile-time draaien. We zullen dan geen Spring hebben. In plaats van Class<?> en Method objecten zullen we objecten uit de Doclet API hebben om de documentatie uit te genereren, zie https://docs.oracle.com/javase/8/docs/jdk/api/javadoc/doclet/index.html. Daarom lijkt het me beter om deze code te houden in plaats van dat we meer Spring te introduceren.

Reply:
Ik snap je punt! Java heeft ook een dergelijke methode `Introspector#getBeanInfo(Class<?> beanClass)` waarna je `getPropertyDescriptors()` kan doen op het object dat de introspector teruggeeft.

Reply:
Die method gebruikt weliswaar geen Spring, maar heeft wel een Class<?> nodig. Als we de complexe code je hier ziet vervangen door het gebruik van Introspector, dan kunnen we straks niet gemakkelijk toe naar compile-time documentatie. Introspector kan geen `ClassDoc` accepteren. De method die ik nu heb gemaakt kunnen we wel simpel omzetten naar`List<FrankAttribute> createAttributes(MethodDoc[] methods, FrankElement attributeOwner)`. Ik heb nog even gezocht naar een Doclet API equivalent van Introspector, maar dat heb ik niet gevonden.